### ISSUE TITLE
Get NSURLErrorCancelled error when try to download one image, but AFNetworking works

### ISSUE THREAD

**Issue by HaleyYQ** [rs/SDWebImage#1921]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

I try to download one image usingAPI sd_setImageWithURL: with the URL "https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg", but it returned the following error. It seems that it doesn't support to download one image if the URL is needed to redirect the url.
But use AFNetworking API setImageWithURL, it can works.

Error Domain=NSURLErrorDomain Code=-999 "cancelled" UserInfo={NSErrorFailingURLKey=https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg, NSLocalizedDescription=cancelled, NSErrorFailingURLStringKey=https://sandbox.dev.clover.com/v2/image/Red_209506569580589592.jpg}

**Comment by mythodeia** [comment]
closing this. you have opened it twice

**Comment by MR-Marsal** [comment]
how to closing, i just use
